### ISSUE TITLE
TypeError when locking a package installed from Git

### ISSUE THREAD

**Issue by ptrstr** [pypa/pipenv#4470]
### Issue description

`pipenv` will throw a `TypeError` while locking. This only occurs when this dependency is added:

```toml
[packages]
sealang = {git = "https://github.com/ptrstr/sealang.git"}
```

Also note this does **not** happen on other Git repositories.

### Expected result

Expected for `pipenv` to lock correctly. As `pip` installs this package from the Git repo without any issues.

### Actual result

```
PS C:\Users\USER\Documents\Projects\Python\PROJECT_NAME> pipenv install --verbose
Using python: 3.7
Path to python: C:/Program Files/Python37/python.exe
Creating a virtualenv for this project…
Pipfile: C:\Users\USER\Documents\Projects\Python\PROJECT_NAME\Pipfile
Using C:/Program Files/Python37/python.exe (3.7.7) to create virtualenv…
[    ] Creating virtual environment...created virtual environment CPython3.7.7.final.0-64 in 502ms
  creator CPython3Windows(dest=C:\Users\USER\.virtualenvs\PROJECT_NAME-KBp1vsq1, clear=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=C:\Users\USER\AppData\Local\pypa\virtualenv)
    added seed packages: pip==20.2.2, setuptools==49.6.0, wheel==0.35.1
  activators BashActivator,BatchActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator

Successfully created virtual environment!
Virtualenv location: C:\Users\USER\.virtualenvs\PROJECT_NAME-KBp1vsq1
Pipfile.lock not found, creating…
Locking [dev-packages] dependencies…
Locking [packages] dependencies…
           Building requirements...
Resolving dependencies...
[   =] Locking...Traceback (most recent call last):
  File "c:/program files/python37/lib/site-packages/pipenv/resolver.py", line 807, in <module>
    main()
  File "c:/program files/python37/lib/site-packages/pipenv/resolver.py", line 803, in main
    parsed.requirements_dir, parsed.packages, parse_only=parsed.parse_only)
  File "c:/program files/python37/lib/site-packages/pipenv/resolver.py", line 785, in _main
Locking Failed!
    resolve_packages(pre, clear, verbose, system, write, requirements_dir, packages)
  File "c:/program files/python37/lib/site-packages/pipenv/resolver.py", line 753, in resolve_packages
    requirements_dir=requirements_dir,
  File "c:/program files/python37/lib/site-packages/pipenv/resolver.py", line 736, in resolve
    req_dir=requirements_dir
  File "c:\program files\python37\lib\site-packages\pipenv\utils.py", line 1386, in resolve_deps
    req_dir=req_dir,
  File "c:\program files\python37\lib\site-packages\pipenv\utils.py", line 1091, in actually_resolve_deps
    deps, index_lookup, markers_lookup, project, sources, req_dir, clear, pre
  File "c:\program files\python37\lib\site-packages\pipenv\utils.py", line 649, in create
    pre=pre, clear=clear
  File "c:\program files\python37\lib\site-packages\pipenv\utils.py", line 463, in get_metadata
    req, resolver=transient_resolver
  File "c:\program files\python37\lib\site-packages\pipenv\utils.py", line 540, in get_deps_from_req
    req_list, lockfile = get_vcs_deps(reqs=[req])
  File "c:\program files\python37\lib\site-packages\pipenv\utils.py", line 1874, in get_vcs_deps
    with temp_path(), locked_repository(requirement) as repo:
  File "c:\program files\python37\lib\contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "c:\program files\python37\lib\site-packages\pipenv\utils.py", line 2030, in locked_repository
    with requirement.req.locked_vcs_repo(src_dir=src_dir) as repo:
  File "c:\program files\python37\lib\contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\requirements.py", line 2210, in locked_vcs_repo
    self._parsed_line.vcsrepo = vcsrepo
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\requirements.py", line 914, in vcsrepo
    kwargs=wheel_kwargs,
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1960, in create
    created.get_initial_info()
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1790, in get_initial_info
    self.update_from_dict(self.parse_setup_py())
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1532, in parse_setup_py
    parsed = ast_parse_setup_py(self.setup_py.as_posix())
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1219, in ast_parse_setup_py
    ast_analyzer.unmap_binops()
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 731, in unmap_binops
    self.binOps_map[binop] = ast_unparse(binop, analyzer=self)
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1030, in ast_unparse
    right_item = unparse(item.right)
    items = unparse(analyzer.assignments[assignment])
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1158, in ast_unparse
    val = unparse(item.value, recurse=False)
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1086, in ast_unparse
    left, ops, right = unparse(item.test)
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1069, in ast_unparse
    if "." in left:
TypeError: argument of type 'Attribute' is not iterable
Traceback (most recent call last):
  File "c:/program files/python37/lib/site-packages/pipenv/resolver.py", line 807, in <module>
    main()
  File "c:/program files/python37/lib/site-packages/pipenv/resolver.py", line 803, in main
    parsed.requirements_dir, parsed.packages, parse_only=parsed.parse_only)
  File "c:/program files/python37/lib/site-packages/pipenv/resolver.py", line 785, in _main
    resolve_packages(pre, clear, verbose, system, write, requirements_dir, packages)
  File "c:/program files/python37/lib/site-packages/pipenv/resolver.py", line 753, in resolve_packages
    requirements_dir=requirements_dir,
  File "c:/program files/python37/lib/site-packages/pipenv/resolver.py", line 736, in resolve
    req_dir=requirements_dir
  File "c:\program files\python37\lib\site-packages\pipenv\utils.py", line 1386, in resolve_deps
    req_dir=req_dir,
  File "c:\program files\python37\lib\site-packages\pipenv\utils.py", line 1091, in actually_resolve_deps
    deps, index_lookup, markers_lookup, project, sources, req_dir, clear, pre
  File "c:\program files\python37\lib\site-packages\pipenv\utils.py", line 649, in create
    pre=pre, clear=clear
  File "c:\program files\python37\lib\site-packages\pipenv\utils.py", line 463, in get_metadata
    req, resolver=transient_resolver
  File "c:\program files\python37\lib\site-packages\pipenv\utils.py", line 540, in get_deps_from_req
    req_list, lockfile = get_vcs_deps(reqs=[req])
  File "c:\program files\python37\lib\site-packages\pipenv\utils.py", line 1874, in get_vcs_deps
    with temp_path(), locked_repository(requirement) as repo:
  File "c:\program files\python37\lib\contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "c:\program files\python37\lib\site-packages\pipenv\utils.py", line 2030, in locked_repository
    with requirement.req.locked_vcs_repo(src_dir=src_dir) as repo:
  File "c:\program files\python37\lib\contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\requirements.py", line 2210, in locked_vcs_repo
    self._parsed_line.vcsrepo = vcsrepo
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\requirements.py", line 914, in vcsrepo
    kwargs=wheel_kwargs,
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1960, in create
    created.get_initial_info()
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1790, in get_initial_info
    self.update_from_dict(self.parse_setup_py())
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1532, in parse_setup_py
    parsed = ast_parse_setup_py(self.setup_py.as_posix())
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1219, in ast_parse_setup_py
    ast_analyzer.unmap_binops()
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 731, in unmap_binops
    self.binOps_map[binop] = ast_unparse(binop, analyzer=self)
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1030, in ast_unparse
    right_item = unparse(item.right)
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1054, in ast_unparse
    items = unparse(analyzer.assignments[assignment])
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1158, in ast_unparse
    val = unparse(item.value, recurse=False)
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1086, in ast_unparse
    left, ops, right = unparse(item.test)
  File "c:\program files\python37\lib\site-packages\pipenv\vendor\requirementslib\models\setup_info.py", line 1069, in ast_unparse
    if "." in left:
TypeError: argument of type 'Attribute' is not iterable
```

### Steps to replicate

```
> pipenv install
```
with this `Pipfile`:
```toml
[[source]]
name = "pypi"
url = "https://pypi.org/simple"
verify_ssl = true

[dev-packages]

[packages]
colorama = "*"
numpy = "*"
sealang = {git = "https://github.com/ptrstr/sealang.git"}

[requires]
python_version = "3.7"
```

-------------------------------------------------------------------------------

<details><summary>$ pipenv --support</summary>

Pipenv version: `'2020.8.13'`

Pipenv location: `'c:\\program files\\python37\\lib\\site-packages\\pipenv'`

Python location: `'c:\\program files\\python37\\python.exe'`

Python installations found:

  - `3.8.2`: `C:\Program Files\Python38\python.exe`
  - `3.7.7`: `C:\Program Files\Python37\python.exe`
  - `2.7.17`: `C:\Python27\python.exe`

PEP 508 Information:

```
{'implementation_name': 'cpython',
 'implementation_version': '3.7.7',
 'os_name': 'nt',
 'platform_machine': 'AMD64',
 'platform_python_implementation': 'CPython',
 'platform_release': '10',
 'platform_system': 'Windows',
 'platform_version': '10.0.19041',
 'python_full_version': '3.7.7',
 'python_version': '3.7',
 'sys_platform': 'win32'}
```

System environment variables:

  - `ALLUSERSPROFILE`
  - `APPDATA`
  - `ASL.LOG`
  - `COMMONPROGRAMFILES`
  - `COMMONPROGRAMFILES(X86)`
  - `COMMONPROGRAMW6432`
  - `COMPUTERNAME`
  - `COMSPEC`
  - `DRIVERDATA`
  - `GOOGLE_API_KEY`
  - `GOOGLE_DEFAULT_CLIENT_ID`
  - `GOOGLE_DEFAULT_CLIENT_SECRET`
  - `GTK_BASEPATH`
  - `HOMEDRIVE`
  - `HOMEPATH`
  - `JAVA_HOME`
  - `LLVM_HOME`
  - `LOCALAPPDATA`
  - `LOGONSERVER`
  - `NUMBER_OF_PROCESSORS`
  - `NVPACK_NDK_TOOL_VERSION`
  - `NVPACK_NDK_VERSION`
  - `NVPACK_ROOT`
  - `ONEDRIVE`
  - `ONEDRIVECONSUMER`
  - `OS`
  - `PATH`
  - `PATHEXT`
  - `PROCESSOR_ARCHITECTURE`
  - `PROCESSOR_IDENTIFIER`
  - `PROCESSOR_LEVEL`
  - `PROCESSOR_REVISION`
  - `PROGRAMDATA`
  - `PROGRAMFILES`
  - `PROGRAMFILES(X86)`
  - `PROGRAMW6432`
  - `PSMODULEPATH`
  - `PUBLIC`
  - `STACK_ROOT`
  - `SYSTEMDRIVE`
  - `SYSTEMROOT`
  - `TEMP`
  - `TMP`
  - `USERDOMAIN`
  - `USERDOMAIN_ROAMINGPROFILE`
  - `USERNAME`
  - `USERPROFILE`
  - `VBOX_MSI_INSTALL_PATH`
  - `VS140COMNTOOLS`
  - `WINDIR`
  - `WSLENV`
  - `WT_PROFILE_ID`
  - `WT_SESSION`
  - `PIP_DISABLE_PIP_VERSION_CHECK`
  - `PYTHONDONTWRITEBYTECODE`
  - `PIP_SHIMS_BASE_MODULE`
  - `PIP_PYTHON_PATH`
  - `PYTHONFINDER_IGNORE_UNSUPPORTED`

Pipenv???specific environment variables:

Debug???specific environment variables:

  - `PATH`: `C:\Program Files\Python37\Scripts\;C:\Program Files\Python37\;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Program Files\Haskell\bin;C:\Program Files\Haskell Platform\8.6.5\lib\extralibs\bin;C:\Program Files\Haskell Platform\8.6.5\bin;C:\ProgramData\Oracle\Java\javapath;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\GtkSharp\2.12\bin;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\dotnet\;C:\Users\USER\.cargo\bin\;C:\cygwin64\bin\;C:\cygwin64\;C:\cygwin64\bin;C:\Users\USER\.cargo\;C:\WINDOWS\System32\OpenSSH\;C:\HaxeToolkit\haxe\;C:\HaxeToolkit\neko;C:\Program Files (x86)\Object.NET\Bridge.NET CLI;C:\Program Files\CMake\bin;C:\Program Files\Haskell Platform\8.6.5\mingw\bin;C:\Program Files\Liquid Technologies\Liquid Studio 2020\XmlDataBinder18\Redist18\cpp\win32\bin;C:\Program Files\Liquid Technologies\Liquid Studio 2020\XmlDataBinder18\Redist18\cpp\win64\bin;C:\Program Files\PuTTY\;C:\Program Files\nodejs\;D:\Program Files\Git\cmd;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\Microsoft Network Monitor 3\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\LLVM\bin;C:\Program Files\MySQL\MySQL Shell 8.0\bin\;C:\Users\USER\AppData\Roaming\cabal\bin;C:\Users\USER\AppData\Roaming\local\bin;C:\Program Files\Arm\Mali Developer Tools\Mali Offline Compiler v6.2.0\;C:\Users\USER\AppData\Local\Microsoft\WindowsApps;C:\MinGW\bin;C:\Users\USER\AppData\Local\GitHubDesktop\bin;C:\Users\USER\AppData\Local\atom\bin;C:\Program Files (x86)\GnuWin32\bin;C:\Users\USER\AppData\Local\Programs\Microsoft VS Code\bin;C:\MinGW\include;C:\Program Files (x86)\Nmap;C:\Program Files\OpenVPN\bin\;C:\Users\USER\AppData\Local\Programs\Fiddler;C:\Users\USER\AppData\Local\Keybase\;C:\Users\USER\AppData\Roaming\npm;C:\Users\USER\AppData\Local\Microsoft\WindowsApps

`

---------------------------

Contents of `Pipfile` ('C:\\Users\\USER\\Documents\\Projects\\Python\\PROJECT_NAME\\Pipfile'):

```toml
[[source]]
name = "pypi"
url = "https://pypi.org/simple"
verify_ssl = true

[dev-packages]

[packages]
colorama = "*"
numpy = "*"
sealang = {git = "https://github.com/ptrstr/sealang.git"}

[requires]
python_version = "3.7"

```

</details>

**Comment by SimonErm** [comment]
Same for https://github.com/Refefer/fastxml

**Comment by SimonErm** [comment]
#4428

**Comment by frostming** [comment]
Fixed on v2021.11.5
